knfc/knfconverter.js

```diff
@@ -0,0 +1,220 @@
+// Generated JS from Java: Node -----
+function Node() {
+    this.left = null;
+    this.right = null;
+    this.parent = null;
+    this.s = null;
+    this.id = null;
+ 
+    if (arguments.length == 0) {
+       jv_Object.call(this);
+    this._NodeInit();
+    }
+    else if (arguments.length == 1) {
+       var s = arguments[0];
+       jv_Object.call(this);
+    this._NodeInit();
+       this.s = s;
+    }}
+ 
+ var Node_c = sc_newClass("Node", Node, jv_Object, null);
+ 
+ Node_c.toString = function ()  {
+    if (this.hasOwnProperty("$protoName")) {
+       return jv_Class_c.toString.apply(this, arguments);
+    }
+    return "[" + (this.s !== null ? this.s : "") + (this.id !== null ? " id=" + this.id : "") + (this.left !== null ? (" : " + this.left.toString() + (this.right !== null ? ("," + this.right.toString()) : "")) : "") + "]";
+ };
+ Node_c.setIDRec = function (id)  {
+    this.id = id;
+    if (this.left !== null) {
+       this.left.setIDRec(id + "0");
+    }
+    if (this.right !== null) {
+       this.right.setIDRec(id + "1");
+    }
+ };
+ Node_c.setID = function (id)  {
+    this.id = id;
+    jv_System_c.out.println("ID SET FOR" + Node_c.toString.call(this));
+ };
+ 
+ Node_c._NodeInit = function() {
+    this.left = null;
+       ;
+    this.right = null;
+       ;
+    this.parent = null;
+       ;
+    this.s = null;
+       ;
+    this.id = null;
+       ;
+ };
+ 
+ // Generated JS from Java: Abfahrt2 -----
+ function Abfahrt2() {
```

Comment:
Genau das gleiche mit der Klasse hier auch wieder, also bitte in eine normale JS Klasse umwandeln

Reply:
Ah und Abfahrt ist zwar ein tolles Wort, jedoch beschreibt dieses nicht was diese Klasse tatsächlich macht